app/addons/fauxton/components.react.jsx

```diff
@@ -269,12 +269,63 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
   });
 
 
+  // A super-simple replacement for window.confirm()
+  var ConfirmationModal = React.createClass({
+    propTypes: {
+      visible: React.PropTypes.bool.isRequired,
+      text: React.PropTypes.string.isRequired,
+      onClose: React.PropTypes.func.isRequired,
+      onSubmit: React.PropTypes.func.isRequired
```

Comment:
In this case, absolutely. We can get a lot of utility out a generic confirmation modal like this. Tying it to a specific store would prevent being used elsewhere.